crypto/property/property.c

```diff
@@ -324,27 +324,31 @@ int ossl_method_store_fetch(OSSL_METHOD_STORE *store, int nid,
     }
 
     if (prop_query == NULL) {
-        if ((impl = sk_IMPLEMENTATION_value(alg->impls, 0)) != NULL) {
-            best_method = &impl->method;
-            ret = 1;
+        if (store->global_properties == NULL) {
+            if ((impl = sk_IMPLEMENTATION_value(alg->impls, 0)) != NULL) {
+                best_method = &impl->method;
+                ret = 1;
+            }
+            goto fin;
```

Comment:
Having this bit smack in the middle of property query processing makes for a confusing read, at least to me.  May I suggest this rearrangement?

``` C
    if (prop_query != NULL) {
        p2 = pq = ossl_parse_query(store->ctx, prop_query);
    }
    if (store->global_properties != NULL) {
        if (pq == NULL) {
            pq = store->global_properties;
        } else {
            p2 = ossl_property_merge(pq, store->global_properties);
            if (p2 == NULL)
                goto fin;
            ossl_property_free(pq);
            pq = p2;
        }
    }

    if (pq == NULL) {
        if ((impl = sk_IMPLEMENTATION_value(alg->impls, 0)) != NULL) {
            best_method = &impl->method;
            ret = 1;
        }
        goto fin;
    }
    optional = ossl_property_has_optional(pq);
    for (j = 0; j < sk_IMPLEMENTATION_num(alg->impls); j++) {
```
```